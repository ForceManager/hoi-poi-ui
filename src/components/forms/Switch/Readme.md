States:

```jsx
<div>
    <Switch size="small" checked />
    <br />
    <Switch size="medium" />
    <br />
    <Switch size="large" checked />
    <br />
    <br />
    <Switch size="small" checked isDisabled />
    <br />
    <Switch size="medium" isDisabled />
    <br />
    <Switch size="large" checked isDisabled />
</div>
```

Interactive:

```jsx
import { useState } from 'react';

const [state, setState] = useState({});

function toggleState(st) {
    if (state.checked) {
        setState({
            checked: false,
        });
    } else {
        setState({
            checked: true,
        });
    }
}

<div>
    <Switch size="small" onClick={toggleState} onChange={toggleState} checked={state.checked} />
</div>;
```

### Component tree

---

-   root - root element
-   [Switch](https://github.com/markusenglund/react-switch)
