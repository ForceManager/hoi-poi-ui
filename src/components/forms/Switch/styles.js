export default (theme) => ({
    root: {
        height: 20,
    },
});
