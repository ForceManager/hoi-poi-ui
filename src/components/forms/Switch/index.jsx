import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import classnames from 'classnames';
import ReactSwitch from 'react-switch';

import { getOverrides } from '../../../utils/overrides';

import styles from './styles';

function Switch({
    checked,
    classes,
    theme,
    className: classNameProp,
    isDisabled,
    onChange,
    overrides: overridesProp,
    ...props
}) {
    // Overrides
    const override = getOverrides(overridesProp, Switch.overrides);

    // Classes
    const rootClassName = classnames(
        classes.root,
        {
            [classes.disabled]: isDisabled,
        },
        classNameProp,
    );

    const rootProps = {
        className: rootClassName,
        onClick: !isDisabled ? onChange : undefined,
    };

    return (
        <div {...rootProps} {...override.root}>
            <ReactSwitch
                className={classes.Switch}
                onChange={!isDisabled ? onChange : undefined}
                checked={checked ? 'checked' : ''}
                disabled={isDisabled}
                offColor={theme.colors.placeholders}
                onColor={theme.colors.primaryDark}
                offHandleColor={theme.colors.greyLight}
                onHandleColor={theme.colors.primary}
                boxShadow="none"
                activeBoxShadow="none"
                handleDiameter={20}
                width={32}
                height={15}
                uncheckedIcon={false}
                checkedIcon={false}
                {...override.Switch}
            />
        </div>
    );
}

Switch.overrides = ['root', 'Switch'];

Switch.defaultProps = {
    checked: false,
    onChange: () => {},
    overrides: {},
};

Switch.propTypes = {
    checked: PropTypes.bool,
    className: PropTypes.string,
    isDisabled: PropTypes.bool,
    onChange: PropTypes.func,
    overrides: PropTypes.object,
};

export default React.memo(withStyles(styles, { name: 'Switch', injectTheme: true })(Switch));
