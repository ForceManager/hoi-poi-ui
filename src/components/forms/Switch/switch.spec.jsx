import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import Switch from './index';

describe('Switch', () => {
    test('unchecked match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Switch />
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('checked match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Switch checked />
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('disabled match snapshot', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Switch isDisabled />
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
