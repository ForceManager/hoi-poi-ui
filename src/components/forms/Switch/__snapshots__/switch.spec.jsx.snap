// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch checked match snapshot 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
        "utils": Object {
          "scrollbar": Object {
            "&::-webkit-scrollbar": Object {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.3.0",
        }
      }
    >
      <Switch
        checked={true}
        onChange={[Function]}
        overrides={Object {}}
      >
        <div
          className="HoiPoi__Switch__root-2-0"
        >
          <ReactSwitch
            activeBoxShadow="none"
            boxShadow="none"
            checked={true}
            checkedIcon={false}
            disabled={false}
            handleDiameter={20}
            height={15}
            offColor="#e1e4e7"
            offHandleColor="#f0f3f5"
            onChange={[Function]}
            onColor="#C87600"
            onHandleColor="#FF8C00"
            uncheckedIcon={false}
            width={32}
          >
            <div
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "MozUserSelect": "none",
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "WebkitTransition": "opacity 0.25s",
                  "WebkitUserSelect": "none",
                  "borderRadius": 7.5,
                  "direction": "ltr",
                  "display": "inline-block",
                  "msUserSelect": "none",
                  "opacity": 1,
                  "position": "relative",
                  "textAlign": "left",
                  "touchAction": "none",
                  "transition": "opacity 0.25s",
                  "userSelect": "none",
                }
              }
            >
              <div
                className="react-switch-bg"
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "MozTransition": "background 0.25s",
                    "WebkitTransition": "background 0.25s",
                    "background": "#C87600",
                    "borderRadius": 7.5,
                    "cursor": "pointer",
                    "height": 15,
                    "margin": 2.5,
                    "position": "relative",
                    "transition": "background 0.25s",
                    "width": 32,
                  }
                }
              />
              <div
                className="react-switch-handle"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "background": "#FF8C00",
                    "border": 0,
                    "borderRadius": "50%",
                    "boxShadow": "none",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "height": 20,
                    "outline": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateX(17px)",
                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "width": 20,
                  }
                }
              />
              <input
                checked={true}
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                role="switch"
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "width": 1,
                  }
                }
                type="checkbox"
              />
            </div>
          </ReactSwitch>
        </div>
      </Switch>
    </JssProvider>
  </ThemeProvider>
</Component>
`;

exports[`Switch disabled match snapshot 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
        "utils": Object {
          "scrollbar": Object {
            "&::-webkit-scrollbar": Object {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.3.0",
        }
      }
    >
      <Switch
        checked={false}
        isDisabled={true}
        onChange={[Function]}
        overrides={Object {}}
      >
        <div
          className="HoiPoi__Switch__root-3-0 undefined"
        >
          <ReactSwitch
            activeBoxShadow="none"
            boxShadow="none"
            checked={false}
            checkedIcon={false}
            disabled={true}
            handleDiameter={20}
            height={15}
            offColor="#e1e4e7"
            offHandleColor="#f0f3f5"
            onChange={[Function]}
            onColor="#C87600"
            onHandleColor="#FF8C00"
            uncheckedIcon={false}
            width={32}
          >
            <div
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "MozUserSelect": "none",
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "WebkitTransition": "opacity 0.25s",
                  "WebkitUserSelect": "none",
                  "borderRadius": 7.5,
                  "direction": "ltr",
                  "display": "inline-block",
                  "msUserSelect": "none",
                  "opacity": 0.5,
                  "position": "relative",
                  "textAlign": "left",
                  "touchAction": "none",
                  "transition": "opacity 0.25s",
                  "userSelect": "none",
                }
              }
            >
              <div
                className="react-switch-bg"
                onClick={null}
                onMouseDown={[Function]}
                style={
                  Object {
                    "MozTransition": "background 0.25s",
                    "WebkitTransition": "background 0.25s",
                    "background": "#e1e4e7",
                    "borderRadius": 7.5,
                    "cursor": "default",
                    "height": 15,
                    "margin": 2.5,
                    "position": "relative",
                    "transition": "background 0.25s",
                    "width": 32,
                  }
                }
              />
              <div
                className="react-switch-handle"
                onClick={[Function]}
                onMouseDown={null}
                onTouchCancel={null}
                onTouchEnd={null}
                onTouchMove={null}
                onTouchStart={null}
                style={
                  Object {
                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "background": "#f0f3f5",
                    "border": 0,
                    "borderRadius": "50%",
                    "boxShadow": "none",
                    "cursor": "default",
                    "display": "inline-block",
                    "height": 20,
                    "outline": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateX(0px)",
                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "width": 20,
                  }
                }
              />
              <input
                checked={false}
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                role="switch"
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "width": 1,
                  }
                }
                type="checkbox"
              />
            </div>
          </ReactSwitch>
        </div>
      </Switch>
    </JssProvider>
  </ThemeProvider>
</Component>
`;

exports[`Switch unchecked match snapshot 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
        "utils": Object {
          "scrollbar": Object {
            "&::-webkit-scrollbar": Object {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.3.0",
        }
      }
    >
      <Switch
        checked={false}
        onChange={[Function]}
        overrides={Object {}}
      >
        <div
          className="HoiPoi__Switch__root-1-0"
        >
          <ReactSwitch
            activeBoxShadow="none"
            boxShadow="none"
            checked={false}
            checkedIcon={false}
            disabled={false}
            handleDiameter={20}
            height={15}
            offColor="#e1e4e7"
            offHandleColor="#f0f3f5"
            onChange={[Function]}
            onColor="#C87600"
            onHandleColor="#FF8C00"
            uncheckedIcon={false}
            width={32}
          >
            <div
              style={
                Object {
                  "MozTransition": "opacity 0.25s",
                  "MozUserSelect": "none",
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "WebkitTransition": "opacity 0.25s",
                  "WebkitUserSelect": "none",
                  "borderRadius": 7.5,
                  "direction": "ltr",
                  "display": "inline-block",
                  "msUserSelect": "none",
                  "opacity": 1,
                  "position": "relative",
                  "textAlign": "left",
                  "touchAction": "none",
                  "transition": "opacity 0.25s",
                  "userSelect": "none",
                }
              }
            >
              <div
                className="react-switch-bg"
                onClick={[Function]}
                onMouseDown={[Function]}
                style={
                  Object {
                    "MozTransition": "background 0.25s",
                    "WebkitTransition": "background 0.25s",
                    "background": "#e1e4e7",
                    "borderRadius": 7.5,
                    "cursor": "pointer",
                    "height": 15,
                    "margin": 2.5,
                    "position": "relative",
                    "transition": "background 0.25s",
                    "width": 32,
                  }
                }
              />
              <div
                className="react-switch-handle"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchCancel={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "background": "#f0f3f5",
                    "border": 0,
                    "borderRadius": "50%",
                    "boxShadow": "none",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "height": 20,
                    "outline": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateX(0px)",
                    "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                    "width": 20,
                  }
                }
              />
              <input
                checked={false}
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                role="switch"
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "width": 1,
                  }
                }
                type="checkbox"
              />
            </div>
          </ReactSwitch>
        </div>
      </Switch>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
